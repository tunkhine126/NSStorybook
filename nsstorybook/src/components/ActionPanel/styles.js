import { makeStyles } from '@material-ui/core/styles';
import colors from 'global-styles';

export const actionPanelStyles = makeStyles(theme => ({
  cancelBtn: {
    color: colors.TEXT.cancel,
    marginTop: 10,
    '&:hover': {
      color: colors.STATUS.negative,
      backgroundColor: 'transparent',
    },
  },
  panel: {
    background: '#D7D7D7',
    padding: '25px 40px',
    height: '100%',
    [theme.breakpoints.down('md')]: {
      padding: '25px 35px',
    },
  },
  confirmBtn: {
    minHeight: 48,
    width: '100%',
    maxWidth: 225,
    boxShadow: 'none',
    borderRadius: 2,
    fontFamily: 'Barlow',
  },
  progress: {
    color: colors.INTERFACE.ui1,
    padding: 8,
    marginTop: -3,
  },
}));
