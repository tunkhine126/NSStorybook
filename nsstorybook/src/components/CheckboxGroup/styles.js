import { makeStyles } from '@material-ui/core/styles';
import styled from 'styled-components';
import colors from 'global-styles';

export const styles = makeStyles(() => ({
  caption: {
    color: colors.INTERFACE.ui5,
    fontSize: 11,
    fontWeight: 500,
  },
  listItemChkbox: {
    marginLeft: -9,
  },
  listItemText: {
    margin: 0,
  },
  resultsCaption: {
    color: colors.TEXT.light,
  },
}));

export const listItemTextStyles = (multi = false) =>
  makeStyles(() => ({
    primary: {
      color: colors.INTERFACE.ui7,
      fontSize: 12,
      fontWeight: 'normal',
    },
    root: {
      marginTop: 2,
      maxWidth: !multi ? 'auto' : 149,
    },
  }));

export const CheckBoxWrapper = styled.span`
  .MuiCheckbox-root {
    height: 22px;
  }
  .MuiCheckbox-root .MuiIconButton-label:hover {
    width: auto;
  }
`;
