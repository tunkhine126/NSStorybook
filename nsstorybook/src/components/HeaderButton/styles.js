import { makeStyles } from '@material-ui/core/styles';
import colors from '../../global-styles';
import styled from 'styled-components';
import { darken } from 'polished';

export const headerButtonStyles = (bgColor = colors.TEXT.medium, padding) =>
  makeStyles(() => ({
    btn: {
      padding,
      marginLeft: '15px',
      fontSize: '12px',
      boxShadow: 'none',
      backgroundColor: bgColor,
      color: colors.INTERFACE.ui1,
      '&:hover': {
        backgroundColor: darken(0.1, bgColor),
      },
    },
    customInput: {
      color: colors.LINKS.link,
      textTransform: 'uppercase',
      fontSize: '12px',
      fontWeight: 600,
      letterSpacing: '1px',
      padding: 0,
      float: 'right',
      borderRadius: 2,
      '&:hover': {
        backgroundColor: 'transparent',
        color: colors.LINKS.hover,
      },
    },
    dropDown: {
      paddingLeft: '10px',
      paddingRight: '10px',
    },
    downArrow: {
      marginLeft: '4px',
      marginBottom: '2px',
    },
    moreBtn: {
      marginLeft: '15px',
      fontSize: '12px',
    },
  }));

export const SelectBtnStyle = styled.span`
  .header-button .MuiSelect-root {
    padding: 8px;
  }
  .header-button:hover {
    background-color: transparent;
  }
  .header-button.Mui-focused {
    background-color: none;
    color: ${colors.LINKS.active};
    border-radius: 2px;
  }
`;
