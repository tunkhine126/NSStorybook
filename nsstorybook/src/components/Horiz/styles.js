import { makeStyles } from '@material-ui/core/styles';
import colors from '../../global-styles';

export const styles = makeStyles(() => ({
  columnHeight: {
    height: 35,
  },
  horiz: {
    color: colors.TEXT.light,
    fontSize: 23,
    '&:hover': {
      color: colors.LINKS.link,
      transition: 'color 0.2s',
    },
  },
}));
