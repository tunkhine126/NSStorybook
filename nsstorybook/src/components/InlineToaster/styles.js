import { makeStyles } from '@material-ui/core/styles';
import styled from 'styled-components';
import colors from 'global-styles';

export const inlineToasterStyle = makeStyles(theme => ({
  success: {
    backgroundColor: colors.STATUS.positive,
    minWidth: 'auto',
    width: 87,
    height: 57,
    padding: '0 5px',
    boxShadow: '0px 2px 5px -1px rgba(138,138,138,1)',
    borderRadius: 2,
    fontWeight: '700',
    fontSize: 15,
    '& svg': {
      fontSize: 20,
    },
  },
  error: {
    backgroundColor: colors.STATUS.negative,
  },
  info: {
    backgroundColor: theme.palette.primary.main,
  },
  warning: {
    backgroundColor: colors.STATUS.notice,
  },
  icon: {
    fontSize: 20,
  },
  iconVariant: {
    opacity: 0.9,
    marginRight: theme.spacing(1),
  },
  message: {
    display: 'flex',
    alignItems: 'center',
  },
  undoBtn: {
    '.MuiButton-root&:hover': {
      color: '#fff',
      opacity: 1,
      transition: 'opacity 0.3s',
    },
  },
}));

export const InlineToasterWrapper = styled.span`
  button {
    color: #fff;
    opacity: 0.7;
    margin-left: 16px;
    font-size: 11px;
  }
  .MuiSnackbarContent-action {
    margin: -10px 0 0px 0;
    padding: 0;
    width: 100%;
    display: flex;
    justify-content: center;
  }
  .MuiSnackbarContent-message {
    padding: 0;
  }
`;
