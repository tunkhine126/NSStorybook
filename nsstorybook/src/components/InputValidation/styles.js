import { makeStyles } from '@material-ui/core/styles';

export const styles = maxWidth =>
  makeStyles(() => ({
    input: {
      width: '100%',
      maxWidth,
    },
  }));
