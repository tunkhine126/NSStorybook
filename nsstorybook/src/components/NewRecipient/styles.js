import { makeStyles } from '@material-ui/core/styles';
import { green, amber } from '@material-ui/core/colors';
import colors from '../../global-styles';
import styled from 'styled-components';

const separatorGray = '#BBBBBB';

export const commonStyles = makeStyles(theme => ({
  darkText: {
    color: colors.TEXT.dark,
    fontSize: '14px',
    fontWeight: 'normal',
  },
  generalContainer: {
    minHeight: '310px',
    borderBottom: `1px solid ${separatorGray}`,
  },
  header: {
    fontSize: '28px',
  },
  input: {
    width: '100%',
    maxWidth: '400px',
  },
  fieldsContainer: {
    padding: '25px 0',
    flexGrow: 0,
  },
  root: {
    flexGrow: 1,
    display: 'flex',
    marginTop: '1.5rem',
    borderTop: `1px solid ${separatorGray}`,
  },
  sectionHeader: {
    fontSize: '16px',
    fontWeight: 'bold',
    color: colors.INTERFACE.ui7,
  },
  spacing: {
    marginBottom: '5px',
  },
  selectSpacing: {
    marginBottom: '27px',
  },
  success: {
    backgroundColor: green[600],
  },
  error: {
    backgroundColor: theme.palette.error.dark,
  },
  info: {
    backgroundColor: theme.palette.primary.main,
  },
  warning: {
    backgroundColor: amber[700],
  },
  linkStyle: {
    textTransform: 'uppercase',
    fontWeight: '700',
    fontSize: '12px',
    letterSpacing: '1px',
  },
  optionalFields: {
    paddingTop: '25px',
  },
  required: {
    minHeight: '202px',
    borderBottom: '1px solid #BBBBBB',
    [theme.breakpoints.down('md')]: {
      minHeight: '225px',
    },
  },
}));

export const InputWrapper = styled.div`
  .MuiInputBase-root {
    height: 58px;
  }
  .MuiFormControl-root {
    max-width: 66%;
  }
  .MuiFilledInput-multiline {
    height: 100px;
  }
`;
