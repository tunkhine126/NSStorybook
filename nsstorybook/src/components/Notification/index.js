/**
 *
 * Notification
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import clsx from 'clsx';

import IconButton from '@material-ui/core/IconButton';
import SnackbarContent from '@material-ui/core/SnackbarContent';
import Snackbar from '@material-ui/core/Snackbar';
import Grid from '@material-ui/core/Grid';

import CloseIcon from '@material-ui/icons/Close';
import ErrorIcon from '@material-ui/icons/Error';
import WarningIcon from '@material-ui/icons/Warning';
import InfoIcon from '@material-ui/icons/Info';
import CheckCircleIcon from '@material-ui/icons/CheckCircle';

import { makeStyles } from '@material-ui/core/styles';

const notificationStyles = makeStyles(() => ({
  margin: {
    marginRight: '10px',
  },
}));

function Notification({
  status,
  message,
  classes,
  messageHandler,
  duration,
  position,
  type,
}) {
  const variantIcon = {
    success: CheckCircleIcon,
    warning: WarningIcon,
    error: ErrorIcon,
    info: InfoIcon,
  };
  const Icon = variantIcon[type];
  const notifyClass = notificationStyles();

  if (!type) {
    return null;
  }

  return (
    <Snackbar
      anchorOrigin={{
        vertical: position.vertical,
        horizontal: position.horizontal,
      }}
      open={status}
      autoHideDuration={duration}
      onClose={messageHandler}
    >
      <SnackbarContent
        className={clsx(classes[type])}
        aria-describedby={`${type}-snackbar`}
        open={false}
        message={
          <Grid
            container
            direction="row"
            justify="center"
            alignItems="center"
            id={`${type}-snackbar`}
            className={classes.message}
          >
            <Icon
              className={clsx(
                classes.icon,
                classes.iconVariant,
                notifyClass.margin
              )}
            />
            {message}
          </Grid>
        }
        action={[
          <IconButton
            key="close"
            aria-label="Close"
            color="inherit"
            onClick={messageHandler}
          >
            <CloseIcon className={classes.icon} />
          </IconButton>,
        ]}
      />
    </Snackbar>
  );
}

Notification.propTypes = {
  status: PropTypes.bool.isRequired,
  message: PropTypes.any,
  classes: PropTypes.object.isRequired,
  messageHandler: PropTypes.func.isRequired,
  duration: PropTypes.number.isRequired,
  position: PropTypes.object.isRequired,
  type: PropTypes.oneOf(['success', 'warning', 'error', 'info']),
};

export default Notification;
