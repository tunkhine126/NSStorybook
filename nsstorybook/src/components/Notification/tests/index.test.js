/**
 *
 * Tests for Notification
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { render } from '@testing-library/react';
import { withStyles } from '@material-ui/core/styles';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import Notification from '../index';

const style = {
  icon: {
    fontSize: 20,
  },
  iconVariant: {
    opacity: 0.9,
    marginRight: '2px',
  },
  message: {
    display: 'flex',
    alignItems: 'center',
  },
};

const state = {
  open: false,
  message: 'Successful log out',
  type: 'error',
};

const Composer = ({ classes }) => (
  <Notification
    status={state.open}
    message={state.message}
    classes={classes}
    messageHandler={() => {}}
    duration={10000}
    position={{ vertical: 'top', horizontal: 'center' }}
    type={state.type}
  />
);

const NotificationComposition = withStyles(style)(Composer);

Composer.propTypes = {
  classes: PropTypes.object.isRequired,
};

describe('<Notification />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<NotificationComposition />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<NotificationComposition />);
    expect(firstChild).toMatchSnapshot();
  });
});
