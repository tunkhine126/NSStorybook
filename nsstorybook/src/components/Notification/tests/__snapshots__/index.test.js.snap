// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> Should render and match the snapshot 1`] = `null`;
